src/libostree/ostree-repo-commit.c

```diff
@@ -912,8 +912,11 @@ write_content_object (OstreeRepo         *self,
                       GCancellable       *cancellable,
                       GError            **error)
 {
+  g_assert (self != NULL);
+  g_assert (OSTREE_IS_REPO (self));
```

Comment:
Feels like we should keep these checks only at API entrypoints and otherwise assume it is correct. WDYT?

Reply:
Yes, I can drop a few of those on the internal-only functions then.

Reply:
Done.